api_v3/services/PlaylistService.php

```diff
@@ -398,7 +399,23 @@ function executeFromContentAction($playlistType, $playlistContent, $detailed = f
 		return KalturaBaseEntryArray::fromDbArray($entryList, $this->getResponseProfile());
 	}
 
-	protected static function handlePlaylistByType($playlistType, $entryFiltersViaEsearch, $entryFiltersViaSphinx, $partnerId, $pagerSeperateQueries, $pager, $totalResults, $playlistContent)
+	function executeFromContentLogic($playlistType, $playlistContent, $detailed = false, $pager = null, $playlistId)
```

Comment:
why you are not calling executeFromContentLogic without playlist id on line 388 instead of duplicating the code?